clang/include/clang/Sema/Sema.h

```diff
@@ -1909,7 +1909,19 @@ class Sema final : public SemaBase {
   /// '\#pragma clang attribute push' directives to the given declaration.
   void AddPragmaAttributes(Scope *S, Decl *D);
 
-  void PrintPragmaAttributeInstantiationPoint();
+  using DiagFuncRef =
+      llvm::function_ref<void(SourceLocation, PartialDiagnostic)>;
+  auto getDefaultDiagFunc() {
+    return [this](SourceLocation Loc, PartialDiagnostic PD) {
+      DiagnosticBuilder Builder(Diags.Report(Loc, PD.getDiagID()));
```

Comment:
Could this just be `this->Diag(Loc, PD);` ?? Or is there a reason for a new Diag-builder/emit call?  It looks like you're capturing `Sema` anyway.

Reply:
This bypasses a lot of the things `this->Diag(Loc, PD)` does, like SFINAE handling, ignoring notes attached to ignored diagnostics, adding instantiation context notes, and more, because we have already done them before storing the suppressed diagnostics, and we are just attaching notes to them.

Reply:
ah!  Thanks for the clarifiation.  Can we get a comment to that effect?